src/js/modules/setup/controllers/change-password/change-password.ctrl.js

```diff
@@ -0,0 +1,305 @@
+
+            if($scope.form.newPassword !== $scope.form.newPasswordRepeat) {
+                $scope.form.passwordCheck = {score: -1};
+                return $q.when(false);
+            }
+
+            if (!$scope.form.newPassword || !$scope.form.newPasswordRepeat) {
+                $scope.form.passwordCheck = null;
+                return $q.when(false);
+            }
+
+            return PasswordStrength.check($scope.form.newPassword, [$scope.form.email, "BTC.com", "wallet"])
+                .then(function(result) {
+                    result.duration = $filter("duration")(result.crack_times_seconds.online_no_throttling_10_per_second * 1000);
+                    $scope.form.passwordCheck = result;
+                    return result;
+                });
+        };
+
+        $scope.changeWalletVersion = function (version) {
+            $scope.form.walletVersion = version;
+        };
+
+        function generateBackupPageTwo (identifier, newEncryptedWalletSecretMnemonic) {
+            return dialogService.alert({
+                title: $translate.instant('CHANGE_PASSWORD'),
+                bodyHtml: $sce.trustAsHtml($translate.instant('CHANGE_PASSWORD_BACKUP')),
+                ok: $translate.instant('BACKUP_DOWNLOAD_ADDITIONAL_PAGE_PDF')
+            }).result.then(function () {
+                var backup = new sdkService.BackupGenerator(
+                    identifier,
+                    {
+                        encryptedSecret: newEncryptedWalletSecretMnemonic
+                    },
+                    {},
+                    {
+                        page1: false,
+                        page2: true,
+                        page3: false
+                    }
+                );
+
+                try {
+                    backup.generatePDF(function(err, pdf) {
+                        if (err) {
+                            $log.error(err);
+                            $scope.alert({
+                                title: $translate.instant('ERROR'),
+                                body: "" + err
+                            });
+                        } else {
+                            pdf.save("BlockTrail Additional Recovery Data Sheet - " + identifier + ".pdf");
+
+                            // delete all temp backup info
+                        }
+                    });
+                } catch (error) {
+                    $log.error("Backup generation error", error);
+                }
+            });
+        }
+
+        function askFor2FA() {
+            return dialogService.prompt(
+                $translate.instant("SETUP_LOGIN"),
+                $translate.instant("MSG_MISSING_TWO_FACTOR_TOKEN")
+            ).result.then(function (result) {
+                twoFactorToken = result;
+            });
+        }
+
+        $scope.encryptNewERS = function() {
+
+            var convert = function(s, from, to) {
+                return (new Buffer(s, from)).toString(to);
+            };
```

Comment:
use `blocktrailSDK.convert` (and add `blocktrailSDK` as controller dependancy I think)